packages/components/src/toolbar/toolbar-button/types.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type ToolbarButtonProps = {
+	/**
+	 * Children to be rendered inside the button.
+	 */
+	children?: ReactNode;
+	/**
+	 * An optional class name for the button container.
+	 */
+	containerClassName?: string;
+	/**
+	 * Additional props to be passed alongside props.
+	 */
+	extraProps?: any;
```

Comment:
I am not sure about the purpose of `extraProps` — both `props` and `extraProps` are spread into `Button`.

Reply:
Agreed. The props passing seems fishy overall, especially with how `props.foo` is used for some things. I have to guess that this code has degraded over time.

I'm also not entirely convinced that we should be accepting _all_ Button props, since some of them are probably not appropriate in this specific toolbar context. But perhaps we can revisit this later when there is a more pressing need to improve these typings and APIs.

For posterity, I'll also note that I'm fine with resorting to an `any` type here for the time being. Ideally it should be some variation of `ButtonProps`, but that forces us to deal with the `button`/`anchor` polymorphism to an extent that I'd rather do it as a separate issue. I kind of feel like we should only allow "Button as a `<button>`" inside toolbars, but even that decision requires some more investigation and effort!

Reply:
This actually relates well to the work that @chad1008 is doing on `CircularOptionPicker`'s TypeScript refactor. For now, we've decided to use `ButtonAsButtonProps` (so that we can avoid the `anchor` case), but I'd definitely be interested in your opinion once the PR is ready for review

Reply:
@ciampo @mirka this has been left as `any` for now. Let me know if a different approach should be taken.

Also, I believe I have worked through the suggestions left prior. This should be ready for review again.